src/adminserver/systemcfg/store/encrypt/driver_test.go

```diff
@@ -0,0 +1,82 @@
+/*
+   Copyright (c) 2016 VMware, Inc. All Rights Reserved.
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package encrypt
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type fakeCfgStore struct {
+	cfgs map[string]interface{}
+}
+
+func (f *fakeCfgStore) Name() string {
+	return "fake"
+}
+
+func (f *fakeCfgStore) Read() (map[string]interface{}, error) {
+	return f.cfgs, nil
+}
+
+func (f *fakeCfgStore) Write(cfgs map[string]interface{}) error {
+	f.cfgs = cfgs
+	return nil
+}
+
+type fakeEncryptor struct {
+}
+
+func (f *fakeEncryptor) Encrypt(plaintext string) (string, error) {
```

Comment:
It doesn't matter. We just make sure the Encrypt() or Decrypt() is called by callers. We are not testing the functions of the fakeEncryptor.